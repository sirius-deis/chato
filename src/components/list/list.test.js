import { render, screen } from "@testing-library/react";
import List from "./list";

const list = ["Item 1", "Item 2"];

describe("List component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <List>
        {list.map((item) => (
          <div>{item}</div>
        ))}
      </List>
    );
    expect(container).toMatchSnapshot();
  });
  it("should contain provided items", () => {
    render(
      <List>
        {list.map((item) => (
          <div>{item}</div>
        ))}
      </List>
    );
    expect(screen.getByText(/Item 1/)).toBeInTheDocument();
    expect(screen.getByText(/Item 2/)).toBeInTheDocument();
  });
});
