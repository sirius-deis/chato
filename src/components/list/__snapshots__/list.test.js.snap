// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot 1`] = `
<div>
  <ul
    class="sc-aXZVg bLxpCv"
    direction="column"
  >
    <li
      class="sc-gEvEer iPakib"
    >
      <div>
        Item 1
      </div>
    </li>
    <li
      class="sc-gEvEer iPakib"
    >
      <div>
        Item 2
      </div>
    </li>
  </ul>
</div>
`;
